API/Controllers/SchoolsController.cs

```diff
@@ -0,0 +1,132 @@
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using API.Models;
+using API.Data;
+using API.DTOs;
+
+namespace API.Controllers;
+
+[ApiController]
+[Route("/[controller]")]
+public class SchoolsController : ControllerBase
+{
+    private readonly myDbContext m_DbContext;
+    
+    public SchoolsController(myDbContext i_DbContext) 
+    {
+        m_DbContext = i_DbContext;
+    }
+    
+    [HttpGet]
+    public async Task<ActionResult<string>> GetSchools()
+    {
+        var schools = await m_DbContext.Schools.ToListAsync();
+        return Ok(schools);
+    }
+    
+    [HttpGet("{i_id}")]
+    public async Task<ActionResult<School>> GetSchoolFromId(int i_id)
+    {
+        var school = await m_DbContext.Schools.FindAsync(i_id);
+        return school != null ? Ok(school) : BadRequest("There is no schools with this id!");
+    }
+
+    [HttpGet("name/{i_SchoolName}")]
+    public async Task<ActionResult<School>> GetSchoolFromName(string i_SchoolName)
+    {
+        var school = await m_DbContext.Schools.FirstOrDefaultAsync(school => school.Name == i_SchoolName);
+        return school != null ? Ok(school) : BadRequest("There is no school with name!");
+    }
+
+    [HttpGet("district/{i_DistrictName}")]
+    public async Task<ActionResult<List<School>>> GetSchoolsFromDistrict(int i_DistrictName)
+    {
+        var schools = await m_DbContext.Schools.ToListAsync();
+        List<School> allSchoolsFromDistrict = new List<School>();
+        foreach (var school in schools)
+        {
+            if (school.DistrictId == i_DistrictName)
+            {
+                allSchoolsFromDistrict.Add(school);
+            }
+        }
+        return schools.Count > 0 ? Ok(allSchoolsFromDistrict) : BadRequest("There are no schools in this district!");
+    }
+
+    [HttpPost("addSchool")]
+    public async Task<ActionResult<string>> AddSchool([FromQuery]SchoolDto i_SchoolDto)
+    {
+        School newSchool = new School()
+        {
+            Name = i_SchoolDto.Name,
+            DistrictId = i_SchoolDto.DistrictId,
+            LicenseId = i_SchoolDto.LicenseId,
+            CreatedAt = DateTime.Now.ToString(),
+            UpdatedAt = DateTime.Now.ToString(),
+            ExpiresAt = DateTime.Now.AddYears(1).ToString()
+            
+        };
+        m_DbContext.Schools.Add(newSchool);
+        await m_DbContext.SaveChangesAsync();
+        return Ok("School added successfully!");
+    }
+    
+    [HttpGet("deleteSchool/{i_schoolId}")]
```

Comment:
This function should work with a "safe delete" instead of real delete. 

Instead of removing the data from the database, add ExpiresAt = DateTime.Now to the column of the 
school.

Please remember that in the Get methods you should now consider that some schools are "deleted" and you should return them.